### ISSUE TITLE
`mapIt` still can't be used with `openArray` even after #8543, #8567: fails during bootstrap

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#8577]
@LemonBoy I was trying to revive my old PR https://github.com/nim-lang/Nim/pull/8272 now that #8543 was merged (Make mapIt work on openArray's #8543) however that PR fails: it was actually kinda confusing since it all works locally (ie when using an already bootstrapped compiler), but the problem is in bootstrap phase:

to reproduce:
```
git clone https://github.com/nim-lang/Nim
cd Nim
git fetch origin pull/8272/head:pr_quoteShellCommand
git checkout pr_quoteShellCommand
sh build_all.sh # (the new install script I added a few commits ago; irrelevant to this bug though)
```

```
/private/tmp/d36/Nim/lib/pure/ospaths.nim(635, 18) template/generic instantiation from here
/private/tmp/d36/Nim/lib/pure/collections/sequtils.nim(680, 14) Error: invalid type: 'openarray[string]' for let
```

for posterity: this is what I was adding and that was failing during bootstrap
```nim
proc quoteShellCommand*(args: openArray[string]): string =
    ## Concatenates and quotes shell arguments `args`
    runnableExamples:
      when defined(posix):
        assert quoteShellCommand(["aaa", "", "c d"]) == "aaa '' 'c d'"
      when defined(windows):
        assert quoteShellCommand(["aaa", "", "c d"]) == "aaa \"\" \"c d\""
    # can't use `map` pending https://github.com/nim-lang/Nim/issues/8303
    result = args.mapIt(quoteShell(it)).join(" ")
```

## links
* #8543
* #8567

**Comment by LemonBoy** [comment]
Uh, that was expected and worked around in sequtils. I've made another PR that should hopefully solve all our problems and give `mapIt` (and me :) some deserved rest.
